### PR TITLE
77 page for admin

### PR DESCRIPTION
**Pull request by SilGosker**

### CODE REVIEW

File: lib/backendFetch.ts [hunk 1]
```diff
@@ -8,7 +8,11 @@ export default async function backendFetch<T>(
   data?: any,
   additionalHeaders?: any
 ): Promise<T | string> {
-  const token = await getToken();
+  let token = await getToken();
+  if (token?.startsWith('"') && token?.endsWith('"')) {
```
Comment (SophiaH67) [inline lib/backendFetch.ts]: Shouldn't be needed

File: lib/signin.ts [hunk 2]
```diff
@@ -12,8 +12,13 @@ export default async function signin(email: string, password: string) {
     email: email,
     password: password,
   });
-
-  await setToken(token);
+  try {
+    const error = JSON.parse(token) as BackendError;
+    alert(error.errorMessage.toString());
```
Comment (SophiaH67) [inline lib/signin.ts]: These types of things should be done in frontend. Check https://github.com/team-jasmijn/frontend/blob/fffb948edfba8b210c83edeb8a7a7ff6f8eb3695/screens/SignUpScreen.tsx#L43-L60

File: screens/HomeScreen.tsx [hunk 3]
```diff
@@ -26,11 +29,50 @@ export default function HomeScreen({
...
 
+  const [companies, setCompanies] = useState<User[]>();
+
+  if (!user) {
+    backendFetch<User>('GET', 'account')
+      .then(e => {
+        setUser(e as User);
+      })
+      .catch(console.log);
+  }
```
Comment (SophiaH67) [inline screens/HomeScreen.tsx]: Wrap in a useEffect.
```ts
useEffect(() => {
  // paste this section in here
}, [])
```

Also, make a seperate file in the ``lib`` folder for ``getUser``. That makes it more portable and easier to type propelry.

Example usage code would be:
```ts
getUser().then(setUser).catch(alert)
```

File: screens/HomeScreen.tsx [hunk 4]
```diff
@@ -26,11 +29,50 @@ export default function HomeScreen({
...
+
+  if (!companies) {
+    backendFetch<User[]>('GET', 'company')
+      .then(e => {
+        setCompanies(e as User[]);
+      })
+      .catch(console.log);
+  }
```
Comment (SophiaH67) [inline screens/HomeScreen.tsx]: Same here

File: screens/HomeScreen.tsx [hunk 5]
```diff
@@ -26,11 +29,50 @@ export default function HomeScreen({
...
+
+  if (user?.isAdmin == true) {
```
Comment (SophiaH67) [inline screens/HomeScreen.tsx]: 1. Remove unneeded ``== true``
2. Consider adding a ``if (!user) return <Text>loading...</Text>``

File: screens/HomeScreen.tsx [hunk 6]
```diff
@@ -26,11 +29,50 @@ export default function HomeScreen({
...
+    return (
+      <View style={styles.main}>
+        <TopBar
+          ScreenName={'Welkom'}
```
Comment (SophiaH67) [inline screens/HomeScreen.tsx]: English please

**Review by SophiaH67 — DISMISSED** [review]

**Review by HUSKUT — APPROVED** [review]

**Review by HUSKUT — APPROVED** [review]

**Pull request merged**